libraries/AP_InertialSensor/AP_InertialSensor_Invensensev3.cpp

```diff
@@ -609,13 +669,47 @@ void AP_InertialSensor_Invensensev3::set_filter_and_scaling_icm42670(void)
     register_write(INV3REG_70_ACCEL_CONFIG1, 0x01);
 }
 
+/*
+  set the filter frequencies and scaling for the ICM-456xy
+ */
+void AP_InertialSensor_Invensensev3::set_filter_and_scaling_icm456xy(void)
+{
+    backend_rate_hz = 6400;
```

Comment:
Is this the only speed supported? This is going to be pretty heavy on the backend SPI thread. If the IMU supports other speeds I think you should allow it to support GYRO_RATE

Reply:
reduced rate to 1.6kHz